### PR TITLE
Add dependency for ScareCrow

### LINKED ISSUE #96: AMI fails to build
# 🐛 Summary #

The AMI [fails to build](https://github.com/cisagov/kali-packer/runs/5238407667?check_suite_focus=true).  The `npm` portion of the BloodHound tool fails.

## To reproduce ##

1. Create a prerelease of the Kali AMI from this repo.
1. Watch the build workflow fail.

## Expected behavior ##

The build workflow should succeed.

## Any helpful log output or screenshots ##

```console
2022-02-17T20:57:40Z:     amazon-ebs: TASK [Build BloodHound] ********************************************************
2022-02-17T20:57:40Z:     amazon-ebs: fatal: [default]: FAILED! => {"changed": true, "cmd": ["npm", "run", "linuxbuild"], "delta": "0:00:00.380912", "end": "2022-02-17 20:57:40.928103", "msg": "non-zero return code", "rc": 1, "start": "2022-02-17 20:57:40.547191", "stderr": "npm ERR! Missing script: \"linuxbuild\"\nnpm ERR! \nnpm ERR! To see a list of scripts, run:\nnpm ERR!   npm run\n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     /root/.npm/_logs/2022-02-17T20_57_40_741Z-debug-0.log", "stderr_lines": ["npm ERR! Missing script: \"linuxbuild\"", "npm ERR! ", "npm ERR! To see a list of scripts, run:", "npm ERR!   npm run", "", "npm ERR! A complete log of this run can be found in:", "npm ERR!     /root/.npm/_logs/2022-02-17T20_57_40_741Z-debug-0.log"], "stdout": "", "stdout_lines": []}
```

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request:
- Adds a dependency that is required for full functionality of [the ScareCrow tool](https://github.com/optiv/ScareCrow).
- Fixes a failure in the building of [the BloodHound tool](https://github.com/BloodHoundAD/BloodHound) that is now happening because of some upstream changes.
- Adds some user data to the base AMI to force `sshd` to allow the use of RSA keys.

## 💭 Motivation and context ##

- Resolves cisagov/cool-system-internal#92
- Resolves #96.
- Resolves cisagov/cool-system-internal#105

## 🧪 Testing ##

All automated tests pass.  I also built a new staging AMI with these changes and verified that it functions as expected and indeed resolves cisagov/cool-system-internal#92, #96, and cisagov/cool-system-internal#105.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.
- [x] Build and test a staging AMI.

## ✅ Pre-merge checklist ##

- [x] Finalize version.

## ✅ Post-merge checklist ##

- [x] Create a release.

### CODE REVIEW

**Comment by jsf9k** [comment]
I'm able to build a new staging AMI, but I can't connect to it via Guacamole.  In the Guacamole web page I get a red band across the screen stating that `The remote desktop server encountered an error and has closed the connection. Please try again or contact your system administrator.` The connection retries every fifteen seconds without success.

I added a Debian Desktop instance and a GoPhish instance, and I am able to connect to those, so I think something is different in this latest Kali build.  The `/home/vnc/.vnc/ip-10-224-48-105.env6:5901.log` file on the Kali instance reports the following:
```
Fri Feb 18 18:03:06 2022
 Connections: accepted: 10.224.49.120::45986
 SConnection: Client needs protocol version 3.8
 SConnection: Client requests security type VncAuth(2)
 VNCSConnST:  Server default pixel format depth 24 (32bpp) little-endian rgb888
 VNCSConnST:  Client pixel format depth 24 (32bpp) little-endian rgb888
 VNCSConnST:  Failed to flush remaining socket data on close: write: Broken
              pipe (32)
 VNCSConnST:  closing 10.224.49.120::45986: write: Connection reset by peer
              (104)
 EncodeManager: Framebuffer updates: 1
 EncodeManager:   ZRLE:
 EncodeManager:     Full Colour: 3 rects, 195.84 kpixels
 EncodeManager:                  115.29 KiB (1:6.63574 ratio)
 EncodeManager:   Total: 3 rects, 195.84 kpixels
 EncodeManager:          115.29 KiB (1:6.63574 ratio)
 Connections: closed: 10.224.49.120::45986
 ComparingUpdateTracker: 48.756 kpixels in / 4.76 kpixels out
 ComparingUpdateTracker: (1:10.2429 ratio)
```
On the Guacamole instance, the `guacamole/guacamole:1.4.0` Docker container reports:
```
guacamole       | 18:02:50.336 [http-nio-8080-exec-3] INFO  o.a.g.tunnel.TunnelRequestService - User "guacadmin" connected to connection "1".
guacamole       | Exception in thread "Thread-113" java.lang.IllegalStateException: Message will not be sent because the WebSocket session has been closed
guacamole       |       at org.apache.tomcat.websocket.WsRemoteEndpointImplBase.writeMessagePart(WsRemoteEndpointImplBase.java:441)
guacamole       |       at org.apache.tomcat.websocket.WsRemoteEndpointImplBase.sendMessageBlock(WsRemoteEndpointImplBase.java:314)
guacamole       |       at org.apache.tomcat.websocket.WsRemoteEndpointImplBase.sendMessageBlock(WsRemoteEndpointImplBase.java:254)
guacamole       |       at org.apache.tomcat.websocket.WsRemoteEndpointImplBase.sendString(WsRemoteEndpointImplBase.java:195)
guacamole       |       at org.apache.tomcat.websocket.WsRemoteEndpointBasic.sendText(WsRemoteEndpointBasic.java:37)
guacamole       |       at org.apache.guacamole.websocket.GuacamoleWebSocketTunnelEndpoint.sendInstruction(GuacamoleWebSocketTunnelEndpoint.java:152)
guacamole       |       at org.apache.guacamole.websocket.GuacamoleWebSocketTunnelEndpoint.access$200(GuacamoleWebSocketTunnelEndpoint.java:53)
guacamole       |       at org.apache.guacamole.websocket.GuacamoleWebSocketTunnelEndpoint$2.run(GuacamoleWebSocketTunnelEndpoint.java:253)
guacamole       | 18:02:50.387 [http-nio-8080-exec-7] INFO  o.a.g.tunnel.TunnelRequestService - User "guacadmin" disconnected from connection "1". Duration: 50 milliseconds
```
while the `guacamole/guacd:1.4.0` Docker container reports:
```
guacd           | guacd[7]: INFO:       Connection "$9b8cf15f-3c2b-4a75-b59a-322a6ac8502e" removed.
guacd           | guacd[7]: INFO:       Creating new client for protocol "vnc"
guacd           | guacd[7]: INFO:       Connection ID is "$dfdb3a68-8921-4816-9007-7fa4b43f44ac"
guacd           | guacd[566]: INFO:     Cursor rendering: local
guacd           | guacd[566]: INFO:     User "@902adb72-72e1-42d5-a692-0b73af2081d4" joined connection "$dfdb3a68-8921-4816-9007-7fa4b43f44ac" (1 users now present)
guacd           | guacd[566]: ERROR:    SSH handshake failed.
guacd           | guacd[566]: INFO:     User "@902adb72-72e1-42d5-a692-0b73af2081d4" disconnected (0 users remain)
guacd           | guacd[566]: INFO:     Last user of connection "$dfdb3a68-8921-4816-9007-7fa4b43f44ac" disconnected
guacd           | guacd[7]: INFO:       Connection "$dfdb3a68-8921-4816-9007-7fa4b43f44ac" removed.
```
The `SSH handshake failed` may be the crux of the issue.

I did note that the latest Kali AMI has version `1.12.0` of TigerVNC, while the AMIs based on Debian Bullseye have version `1.11.0`.

**Comment by jsf9k** [comment]
I confirmed via the Docker images `debian:bullseye-slim` and `kalilinux/kali-rolling:latest` that the former has only version `1.11.0` of TigerVNC available:
```
# apt-cache policy tigervnc-standalone-server tigervnc-common
tigervnc-standalone-server:
  Installed: (none)
  Candidate: 1.11.0+dfsg-2
  Version table:
     1.11.0+dfsg-2 500
        500 http://deb.debian.org/debian bullseye/main amd64 Packages
tigervnc-common:
  Installed: (none)
  Candidate: 1.11.0+dfsg-2
  Version table:
     1.11.0+dfsg-2 500
        500 http://deb.debian.org/debian bullseye/main amd64 Packages
```
while the latter has only version `1.12.0` available:
```
# apt-cache policy tigervnc-standalone-server tigervnc-common
tigervnc-standalone-server:
  Installed: (none)
  Candidate: 1.12.0+dfsg-2
  Version table:
     1.12.0+dfsg-2 500
        500 http://http.kali.org/kali kali-rolling/main amd64 Packages
tigervnc-common:
  Installed: (none)
  Candidate: 1.12.0+dfsg-2
  Version table:
     1.12.0+dfsg-2 500
        500 http://http.kali.org/kali kali-rolling/main amd64 Packages
```

**Comment by jsf9k** [comment]
I can see a line like this at the end of the output of `sudo journalctl -e -f --unit ssh.service` every time I try to connect via Guacamole:
```
Feb 18 18:35:21 kali sshd[2464]: Unable to negotiate with 10.224.49.120 port 57868: no matching host key type found. Their offer: ssh-rsa,ssh-dss [preauth]
```

`10.224.49.120` is the internal IP of the Guacamole instance.  I suspect this `ssh` issue is the crux of the problem.

**Comment by jsf9k** [comment]
I was able to get the Guacamole-Kali connection to work by adding this to the Kali instance's `/etc/ssh/sshd_config` instance:
```ssh-config
HostKeyAlgorithms +ssh-rsa
PubkeyAcceptedKeyTypes +ssh-rsa
```

**Comment by jsf9k** [comment]
I created a new staging Kali AMI with the code changes from this PR and was able to verify that `ScareCrow` is now functioning as expected: 
```
$ ./ScareCrow -I /home/vnc/beacon.exe -domain example.com                                                                                                                                                                       
                                                                                                                                                                                                                                           
  _________                           _________                                                                                                                                                                                            
 /   _____/ ____ _____ _______   ____ \_   ___ \_______  ______  _  __                                                                                                                                                                     
 \_____  \_/ ___\\__  \\_  __ \_/ __ \/    \  \/\_  __ \/  _ \ \/ \/ /                                                                                                                                                                     
 /        \  \___ / __ \|  | \/\  ___/\     \____|  | \(  <_> )     /                                                                                                                                                                      
/_______  /\___  >____  /__|    \___  >\______  /|__|   \____/ \/\_/                                                                                                                                                                       
        \/     \/     \/            \/        \/                                                                                                                                                                                           
                                                        (@Tyl0us)                                                                                                                                                                          
        ?Fear, you must understand is more than a mere obstacle. 
        Fear is a TEACHER. the first one you ever had.?

[*] Encrypting Shellcode Using AES Encryption
[+] Shellcode Encrypted
[+] Patched ETW Enabled
[+] Sleep Timer set for 2747 milliseconds 
[*] Creating an Embedded Resource File
[+] Created Embedded Resource File With OneDrive's Properties
[*] Compiling Payload
[+] Payload Compiled
[*] Signing OneDrive.exe With a Fake Cert
[+] Signed File Created
[+] Binary Compiled

$ 
```

**Comment by jsf9k** [comment]
Labeling this PR as `blocked`, since cisagov/cool-system-internal#105 must be resolved before we can roll out a new production AMI based on these changes.

**Review by dav3r — APPROVED** [review]
LGTM!  🌽

**Review by dav3r — APPROVED** [review]
Now even better than before. ⚙️

**Comment by jsf9k** [comment]
Version `0.5.1-rc.3` is working and currently deployed to env-6 staging.

**Comment by jsf9k** [comment]
We had some assessment folks run `apt-get upgrade` on a Kali in a production, which upgraded OpenSSH to a version that exhibits this issue.  @mcdonnnj, can you review this at your earliest convenience?

File: src/user_data.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+# Here we force OpenSSH to allow the use of RSA keys.
+#
+# See https://github.com/cisagov/cool-system-internal/issues/105 for
+# more details.
+
+cat > /etc/ssh/sshd_config.d/99-allow-rsakeys.conf << ENDOFFILE
+HostKeyAlgorithms +ssh-rsa
+PubkeyAcceptedKeyTypes +ssh-rsa
```
Comment (mcdonnnj) [inline src/user_data.sh]: Per the [8.5 release notes](https://www.openssh.com/txt/release-8.5):

> ssh(1), sshd(8): rename the PubkeyAcceptedKeyTypes keyword to
   PubkeyAcceptedAlgorithms. The previous name incorrectly suggested
   that it control allowed key algorithms, when this option actually
   specifies the signature algorithms that are accepted. The previous
   name remains available as an alias. bz#3253

```suggestion
PubkeyAcceptedAlgorithms +ssh-rsa
```

**Review by mcdonnnj — APPROVED** [review]
Makes sense and thanks for fixing these issues 💪. I had one suggestion regarding the OpenSSH configuration change.

**Pull request merged**